src/mixins/filing-template-mixin.ts

```diff
@@ -241,6 +241,19 @@ export default class FilingTemplateMixin extends Mixins(AmalgamationMixin, DateM
 
     // restore business name data
     const nameRequest = draftFiling.amalgamationApplication.nameRequest as NameRequestFilingIF
+
+    if (this.getAmalgamationType !== AmalgamationTypes.REGULAR) {
+      this.setCorrectNameOption(CorrectNameOptions.CORRECT_AML_ADOPT)
+
+      const holdingBusiness = draftFiling.amalgamationApplication.amalgamatingBusinesses.find(
+        (business) => business.role === 'holding'
+      )
+      if (holdingBusiness) {
+        this.fetchAmalgamatingBusinessInfo(holdingBusiness.identifier).then((result) => {
+          this.setNameRequestApprovedName(result.businessInfo.legalName);
+        })
+      }
+    }
```

Comment:
It looks like this will work, but:

1. I still believe the legal name should be pre-populated when the draft amalgamation is created in Filings UI, not "patched" here. This is going back to which code (UI) should be doing what.

2. When you rebase, you will already have the `holdingPrimary` object (if it exists), so you don't need to find it yourself.

3. You should enums instead of strings (line 249).

4. It would be better to use await instead of then when fetching data. When using await, the code stops executing until the fetch is completed, and then continues executing. When using then, its code block is executed after the fetch is complete (in a different time-slice), but the other code in here executes immediately.

5. After you rebase, you won't need to fetch the amalgamating business info for the holding business as it will already have been loaded (per point 2 above).